### PR TITLE
Fix/id mapper

### PR DESCRIPTION
**Pull request by timosachsenberg**
## Description

Fixes TMT IDMapper issue. No with a proper test to catch this issue.

## Checklist
- [ ] Make sure that you are listed in the AUTHORS file
- [ ] Add relevant changes and new features to the CHANGELOG file
- [ ] I have commented my code, particularly in hard-to-understand areas
- [ ] New and existing unit tests pass locally with my changes
- [ ] Updated or added python bindings for changed or new classes (Tick if no updates were necessary.)

### How can I get additional information on failed tests during CI
<details>
  <summary>Click to expand</summary>
If your PR is failing you can check out

- The details of the action statuses at the end of the PR or the "Checks" tab.
- http://cdash.openms.de/index.php?project=OpenMS and look for your PR. Use the "Show filters" capability on the top right to search for your PR number.
  If you click in the column that lists the failed tests you will get detailed error messages.

</details>

### Advanced commands (admins / reviewer only)
<details>
  <summary>Click to expand</summary>
  
- `/reformat` (experimental) applies the clang-format style changes as additional commit. Note: your branch must have a different name (e.g., yourrepo:feature/XYZ) than the receiving branch (e.g., OpenMS:develop). Otherwise, reformat fails to push.
- setting the label "NoJenkins" will skip tests for this PR on jenkins (saves resources e.g., on edits that do not affect tests)
- commenting with `rebuild jenkins` will retrigger Jenkins-based CI builds
  
</details>

---
:warning: Note: Once you opened a PR try to minimize the number of *pushes* to it as every push will trigger CI (automated builds and test) and is rather heavy on our infrastructure (e.g., if several pushes per day are performed).

### CODE REVIEW

File: src/openms/source/ANALYSIS/ID/IDMapper.cpp [hunk 1]
```diff
@@ -362,14 +363,19 @@ namespace OpenMS
...
+            OPENMS_LOG_WARN << "ConsensusMap for TMT/iTRAQ experiment contains scan identifier '" << cf_scan_id 
                           << "' quantified in file '" << filename 
                           << "' but there is no matching identification."
-                          << std::endl;
+                          << std::endl;            
+          }        
+        }
+        else // missing spectrum id annotation
+        {
+            OPENMS_LOG_WARN << "ConsensusMap for TMT/iTRAQ experiment is missing the scan identifier meta value '" << cf_scan_id << "'"
```
Comment (timosachsenberg) [inline src/openms/source/ANALYSIS/ID/IDMapper.cpp:377]:
```suggestion
            OPENMS_LOG_WARN << "ConsensusMap for TMT/iTRAQ experiment is missing the scan identifier meta value '" << cf_scan_id_key_name << "'"
```

**Review by jpfeuffer — COMMENTED** [review]
What changed? What was the problem?

**Comment by timosachsenberg** [comment]
Read wrong meta value. Was definitely a bug.

**Comment by jpfeuffer** [comment]
You were looking for the scan id "controller type..." as metavalue basically, right?

**Review by jpfeuffer — APPROVED** [review]

**Comment by jpfeuffer** [comment]
Unfortunately we will need a new conda release. A new version number. New release tar gz. New hash. New conda PR.

**Comment by timosachsenberg** [comment]
Sorry about that, don't know how this could have been missed. I thought this change was properly “manually” tested by the person that opened an issue. Good to have a proper test now.

**Pull request merged**